infra/base-images/base-builder/compile

```diff
@@ -22,13 +22,10 @@ if [ -z "${SANITIZER_FLAGS-}" ]; then
   export SANITIZER_FLAGS=${!FLAGS_VAR-}
 fi
 
-if [[ $FUZZING_ENGINE != "none" ]]; then
-  # compile script might override environment, use . to call it.
-  . compile_${FUZZING_ENGINE}
-fi
+# We don't need to compile fuzzing engine with sanitizers, unless MSan is used.
+FUZZING_ENGINE_SANITIZER_FLAGS=""
 
-if [[ $SANITIZER_FLAGS = *sanitize=memory* ]]
-then
+if [[ $SANITIZER_FLAGS = *sanitize=memory* ]]; then
   # Take all libraries from lib/msan and MSAN_LIBS_PATH
   # export CXXFLAGS_EXTRA="-L/usr/msan/lib $CXXFLAGS_EXTRA"
   cp -R /usr/msan/lib/* /usr/lib/
```

Comment:
@oliverchang to also comment if this changing of order to compile fuzzing engine later has any implications.